### PR TITLE
[ios] Fix text input edit rotor accessibility

### PR DESCRIPTION
**Pull request by hellohuanlin**
Fixes text input edit rotor accessibility missing edit actions.

It also fixes a few edit items that is displayed where it's not supposed to. (e.g. displaying select item while nothing to select).

*List which issues are fixed by this PR. You must list at least one issue.*

Fixes https://github.com/flutter/flutter/issues/151029

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by hellohuanlin** [comment]
Verified `Select` and `Select All` options will only be available when the text field is not empty.

**Comment by hellohuanlin** [comment]
Verified still need double tap gesture to reveal the edit options, but it seems to be also true for native text field. I will check our behavior before [this change](https://github.com/flutter/engine/pull/52333) to see if it's a regression.

**Comment by hellohuanlin** [comment]
After reverting https://github.com/flutter/engine/pull/52333, I was able to confirm that it's been an existing behavior to require to double tap to bring up this rotor. So it's not a regression.

File: shell/platform/darwin/ios/framework/Source/TextInputSemanticsObject.mm [hunk 1]
```diff
@@ -462,4 +462,34 @@ - (BOOL)hasText {
   return [[self textInputSurrogate] hasText];
 }
 
+#pragma mark - UIResponder overrides
+
+- (void)cut:(id)sender {
```
Comment (jmagman) [inline shell/platform/darwin/ios/framework/Source/TextInputSemanticsObject.mm:467]: I'm not sure all the gotchas, but can this class use `forwardingTargetForSelector` instead of directly calling every method on `textInputSurrogate`? It's inevitable we will miss some as they are added in future SDKs.  I'm not sure it's the right behavior for this class, though, and I'm not sure if the super class `UIAccessibilityElement` already implements, in which case it won't work...

**Comment by hellohuanlin** [comment]
Not related to objc message forwarding, but found a very interesting crash on select. Looking into it!

File: shell/platform/darwin/ios/framework/Source/TextInputSemanticsObject.mm (hunk 1 above)
Comment (hellohuanlin) [reply shell/platform/darwin/ios/framework/Source/TextInputSemanticsObject.mm:467]: It turns out that iOS determines whether to show these options in accessibility rotor by checking the presence of these functions, So we can't use `forwardingTargetForSelector`.

**Comment by hellohuanlin** [comment]
Surprisingly, it turns out that `select:` button is not supported (see https://github.com/flutter/flutter/issues/107578), hence the crash. Adding `select:` would be out of scope of this PR.

**Review by chunhtai — APPROVED** [review]
LGTM

**Comment by luomo-pro** [comment]
@hellohuanlin
Hello, after testing, I found that it's not perfect yet. There are still two issues:

1. After selecting all content, the "select all" operation is still available in the rotor.
2. Operations can only be performed on the text field that has the keyboard focus activated.

As I mentioned in #151029 :

Suppose there are two text fields on an interface.
1. Double-tap to activate the first text field, and the keyboard pops up.
2. Then, use VoiceOver to focus on the second text field. Note: just focus on it, don't double-tap.
3. Finally, use the rotor to perform a paste operation.

Native behavior: The text will be pasted into the second text field.
Flutter behavior: Unable to perform the paste operation.

In other words, the native rotor operation targets the text field that VoiceOver is focused on.
However, Flutter rotor operation only targets the text field that has the keyboard activated.

**Comment by hellohuanlin** [comment]
@luomo-pro 

> After selecting all content, the "select all" operation is still available in the rotor.

I can probably put up a PR to disable selectAll in that case

> Operations can only be performed on the text field that has the keyboard focus activated.

This has been the existing behavior before I reverted the PR, so not a regression. Can you create a new issue for that?

**Comment by luomo-pro** [comment]
@hellohuanlin
Okay, thank you very much. I will submit a separate issue for the second question
